prove/Develop05/Program.cs

```diff
@@ -172,22 +172,20 @@ static void Main(string[] args)
                  }
                  Console.Write("Which goal did you accomplish? ");
                  int answer = int.Parse(Console.ReadLine());
-                 if(goals[answer-1].GetType()== "CheckList Goal")
+                 if(goals[answer-1].GetType()== "CheckList Goal" && goals[answer-1].IsComplete()==false)
```

Comment:
this was really important because this gave the way to make a difference between two or more checklist goals, if I have a checklist that is done, I send a message saying that, but if it is not done, we call the method, and plus the points, then in the method we keep track of how many times they have done and if it is completed to get the bonus.

Reply:
Just let the RecordEvents() method handle that.  Encapsulation indicates that the Program shouldn't need to know that a checklist can't be completed if it is already complete.  That is a detail for the ChecklistGoal class to worry about.